### PR TITLE
roachtest: update regression.diffs

### LINKED ISSUE #132515: roachtest: pg_regress failed
roachtest.pg_regress [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestWeeklyBazel/17287519?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestWeeklyBazel/17287519?buildTab=artifacts#/pg_regress) on master @ [30dbb173d0f083b35cf9eb8093832a5dd764c5af](https://github.com/cockroachdb/cockroach/commits/30dbb173d0f083b35cf9eb8093832a5dd764c5af):

```
-+See: https://go.crdb.dev/issue-v/126359/_version_
++ERROR:  relation "d" does not exist
  INSERT INTO alter_table_under_transition_tables
    VALUES (1, '1'), (2, '2'), (3, '3');
  UPDATE alter_table_under_transition_tables
@@ -238084,13 +237964,11 @@
  --
  -- Test multiple reference to a transition table
  --
-@@ -5764,19 +6504,37 @@
+@@ -5764,19 +6417,35 @@
  CREATE TRIGGER my_trigger AFTER UPDATE ON multi_test
    REFERENCING NEW TABLE AS new_test OLD TABLE as old_test
    FOR EACH STATEMENT EXECUTE PROCEDURE multi_test_trig();
-+ERROR:  unimplemented: CREATE TRIGGER
-+HINT:  You have attempted to use a feature that is not yet implemented.
-+See: https://go.crdb.dev/issue-v/126359/_version_
++ERROR:  relation "new_test" does not exist
  UPDATE multi_test SET i = i;
 -NOTICE:  count = 1
 -NOTICE:  count union = 2
@@ -238124,7 +238002,7 @@
  CREATE OR REPLACE FUNCTION get_from_partitioned_table(partitioned_table.a%type)
  RETURNS partitioned_table AS $$
  DECLARE
-@@ -5787,13 +6545,13 @@
+@@ -5787,13 +6456,13 @@
      SELECT * INTO result FROM partitioned_table WHERE a = a_val;
      RETURN result;
  END; $$ LANGUAGE plpgsql;
@@ -238144,7 +238022,7 @@
  CREATE OR REPLACE FUNCTION list_partitioned_table()
  RETURNS SETOF partitioned_table.a%TYPE AS $$
  DECLARE
-@@ -5806,14 +6564,13 @@
+@@ -5806,14 +6475,13 @@
      END LOOP;
      RETURN;
  END; $$ LANGUAGE plpgsql;
@@ -238165,7 +238043,7 @@
  --
  -- Check argument name is used instead of $n in error message
  --
-@@ -5822,6 +6579,16 @@
+@@ -5822,6 +6490,16 @@
    GET DIAGNOSTICS x = ROW_COUNT;
    RETURN;
  END; $$ LANGUAGE plpgsql;
If the diff is expected, copy regression.diffs from the test artifacts to pkg/cmd/roachtest/testdata/regression.diffs
test artifacts and logs in: /artifacts/pg_regress/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_runtimeAssertionsBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>

See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)

See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-17287519/pg-regress/1728712720168/1728713397127)

</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*pg_regress.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

Jira issue: CRDB-43105

### PR DESCRIPTION
**Pull request by mw5h**
Updated the regression.diffs file to reflect changes due to improved TRIGGER and collation support. Filed issues for the panic and trigger misbehavior.

Fixes: #132515
Informs: #135311
Informs: #135131

Release note: None

### CODE REVIEW

**Review by yuzefovich — COMMENTED** [review]
We should preserve `_version_` markers that were added in #131835.

**Review by mw5h — COMMENTED** [review]
Done.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/135313#-:-OC5GRIFA5k-Hae0vOQu:b-896fix)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @srosenberg and @vidit-bhat)

File: pkg/cmd/roachtest/testdata/regression.diffs [hunk 1]
```diff
@@ -77372,49 +77183,818 @@ diff -U3 --label=/mnt/data1/postgres/src/test/regress/expected/triggers.out --la
...
-+See: https://go.crdb.dev/issue-v/126359/_version_
++ERROR:  no data source matches pattern: old.*
  -- check dependency restrictions
  ALTER TABLE main_table DROP COLUMN b;
 -ERROR:  cannot drop column b of table main_table because other objects depend on it
 -DETAIL:  trigger after_upd_b_row_trig on table main_table depends on column b of table main_table
 -trigger after_upd_a_b_row_trig on table main_table depends on column b of table main_table
 -trigger after_upd_b_stmt_trig on table main_table depends on column b of table main_table
 -HINT:  Use DROP ... CASCADE to drop the dependent objects too.
++ERROR:  error executing StatementPhase stage 1 of 1 with 8 MutationType ops: relation "main_table" (649): trigger "after_upd_b_row_trig" contains unknown column "b"
```
Comment (yuzefovich) [inline pkg/cmd/roachtest/testdata/regression.diffs:77224]: Should we extract this error into a separate issue?

File: pkg/cmd/roachtest/testdata/regression.diffs [hunk 2]
```diff
@@ -78313,9 +78652,52 @@ diff -U3 --label=/mnt/data1/postgres/src/test/regress/expected/triggers.out --la
  AS 'begin RETURN NULL; end';
  CREATE TRIGGER no_op_trig INSTEAD OF INSERT OR UPDATE OR DELETE
  ON european_city_view FOR EACH ROW EXECUTE PROCEDURE no_op_trig_fn();
-+ERROR:  unimplemented: CREATE TRIGGER
-+HINT:  You have attempted to use a feature that is not yet implemented.
-+See: https://go.crdb.dev/issue-v/126359/_version_
++ERROR:  internal error: building declarative schema change targets for CREATE TRIGGER: runtime error: invalid memory address or nil pointer dereference
```
Comment (yuzefovich) [inline pkg/cmd/roachtest/testdata/regression.diffs:78655]: I think this bug has already been addressed in #135667, but we can update the diff on the next failure.

**Review by yuzefovich — APPROVED** [review]
:lgtm: with one question.

**Review by mw5h — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/135313)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @srosenberg, @vidit-bhat, and @yuzefovich)

**Pull request merged**